### ISSUE TITLE
InvalidAccessKeyId from SDK, fine from CLI

### ISSUE THREAD

**Issue by gregtakacs** [aws/aws-sdk-cpp#365]
If I run ' aws s3 ls' from the command prompt things work fine but when I try to run the list_buckets sample code I get the following error:

ListBuckets error: InvalidAccessKeyId Unable to parse ExceptionName: InvalidAccessKeyId Message: The AWS Access Key Id you provided does not exist in our records.

What would be causing this?

If I run 'aws configure list' it shows that my key and ID are coming from an environment variable but I also have the .aws/configure file set with the same information.

**Comment by gregtakacs** [comment]
Problem is solved: in GovCloud you have to override the ClientConfiguration for both **config.region** as well as **config.endpointOverride**

**Comment by JonathanHenson** [comment]
Yes, sorry for the complication. You have to set overrides for gov cloud. We'll be improving this experience in the future.

**Comment by KeithBlonquist** [comment]
I am trying to access Landsat imagery on US-WEST-2 from US-GOV-WEST-1 using the S3Client. I have set the region to US_WEST_2 and I set the endpointOverride to "s3-us-west-2.amazonaws.com". I am getting an InvalidAccessKeyId error. What am I missing? The log reports that a GET request is being made to:

https://s3-us-west-2.amazonaws.com/landsat-pds/L8/038/031/LC80380312015006LGN00/LC80380312015006LGN00_MTL.txt

I can access the file with this URI from outside of AWS.
